plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -802,73 +802,99 @@ func (checker *PodAffinityChecker) InterPodAffinityMatches(pod *api.Pod, meta in
 	if err != nil {
 		return false, err
 	}
-	if checker.NodeMatchPodAffinityAntiAffinity(pod, allPods, node) {
-		return true, nil
+	affinity, err := api.GetAffinityFromPodAnnotations(pod.Annotations)
+	if err != nil {
+		return false, err
+	}
+
+	// Check if the current node match the inter-pod affinity scheduling constraints.
+	// Hard inter-pod affinity is not symmetric, check only when affinity.PodAffinity exists.
+	if affinity.PodAffinity != nil {
+		if !checker.NodeMatchesHardPodAffinity(pod, allPods, node, affinity.PodAffinity) {
+			return false, ErrPodAffinityNotMatch
+		}
+	}
+
+	// Hard inter-pod anti-affinity is symmetric, we should always check it.
+	if !checker.NodeMatchesHardPodAntiAffinity(pod, allPods, node, affinity.PodAntiAffinity) {
+		return false, ErrPodAffinityNotMatch
 	}
-	return false, ErrPodAffinityNotMatch
+
+	return true, nil
 }
 
 // AnyPodMatchesPodAffinityTerm checks if any of given pods can match the specific podAffinityTerm.
-func (checker *PodAffinityChecker) AnyPodMatchesPodAffinityTerm(pod *api.Pod, allPods []*api.Pod, node *api.Node, podAffinityTerm api.PodAffinityTerm) (bool, error) {
+// TODO: Do we really need any pod matching, or all pods matching? I think the latter.
```

Comment:
OK, so what I had in mind is:
- assume that we have a service and it has 2 underlying pods: 1 in zone A and 1 in zone B
- then if the pod we are scheduling is going to talk to this service, its connections will be forwarded to random from those 2 pods (we don't have service affinity, right?)
- then ideally, we would like to be close to both pods, because we don't know which one we will be talking to

However, in the example above, this is impossible, so maybe it is enough to actually be close to "any of those"?